### ISSUE TITLE
grpo训练在第一步卡死

### ISSUE THREAD

**Issue by HongyongZZZ** [modelscope/ms-swift#5678]
grpo.sh  
```
CUDA_VISIBLE_DEVICES=1,2,3 \
NPROC_PER_NODE=3 \
swift rlhf \
    --rlhf_type grpo \
    --model Qwen/Qwen2-Audio-7B-Instruct \
    --model-type qwen2_audio \
    --reward_funcs external_audio_reward \
    --reward_weights 1 \
    --use_vllm true \
    --vllm_mode server \
    --vllm_server_host 0.0.0.0 \
    --vllm_server_port 8000 \
    --train_type full \
    --torch_dtype bfloat16 \
    --dataset /home/user/rl.json \
    --external_plugins examples/train/grpo/plugin/plugin.py \
    --max_completion_length 4096 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 8 \
    --per_device_eval_batch_size 8 \
    --learning_rate 3e-7 \
    --gradient_accumulation_steps 2 \
    --eval_steps 300 \
    --save_steps 300 \
    --save_total_limit 2 \
    --logging_steps 5 \
    --max_length 8192 \
    --output_dir output \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 4 \
    --dataset_num_proc 4 \
    --num_generations 8 \
    --temperature 1. \
    --top_p 0.99 \
    --top_k 50 \
    --deepspeed zero3 \
    --log_completions true \
    --num_iterations 1 \
    --beta 0.001 
```

问题：似乎训练卡死在第一步，之前的3.6dev版本还能正常运行

<img width="1719" height="397" alt="Image" src="https://github.com/user-attachments/assets/1d117cd1-0b10-4623-8653-ce4e05c4f532" />

**Comment by hjh0119** [comment]
1. **Your hardware and system info**
Write your system info like CUDA version/system/GPU/torch/swift version here

2. retry example script

**Comment by HongyongZZZ** [comment]
> 1. **Your hardware and system info**
>    Write your system info like CUDA version/system/GPU/torch/swift version here
> 2. retry example script

**Thank you for your reply. My hardware information and script are as follows：**

**1. CUDA: 12.6，system: Linux，GPU：4*A100，torch version：2.7.1，swift version: 3.8.0.dev0**
 
**2. Async(External) Mode
server:** 
`CUDA_VISIBLE_DEVICES=0 swift rollout  --model Qwen/Qwen2-Audio-7B-Instruct --model_type qwen2_audio`

**client:** 
```
CUDA_VISIBLE_DEVICES=1,2,3 \
NPROC_PER_NODE=3 \
swift rlhf \
    --rlhf_type grpo \
    --model Qwen/Qwen2-Audio-7B-Instruct \
    --model-type qwen2_audio \
    --reward_funcs external_audio_reward \
    --reward_weights 1 \
    --use_vllm true \
    --vllm_mode server \
    --vllm_server_host 0.0.0.0 \
    --vllm_server_port 8000 \
    --train_type full \
    --torch_dtype bfloat16 \
    --dataset /home/user/rl.json \
    --external_plugins examples/train/grpo/plugin/plugin.py \
    --max_completion_length 4096 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 8 \
    --per_device_eval_batch_size 8 \
    --learning_rate 3e-7 \
    --gradient_accumulation_steps 2 \
    --eval_steps 300 \
    --save_steps 300 \
    --save_total_limit 2 \
    --logging_steps 5 \
    --max_length 8192 \
    --output_dir output \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 4 \
    --dataset_num_proc 4 \
    --num_generations 8 \
    --temperature 1. \
    --top_p 0.99 \
    --top_k 50 \
    --deepspeed zero3 \
    --log_completions true \
    --num_iterations 1 \
    --beta 0.001 
```

**Comment by hjh0119** [comment]
Can you retry by upgrading Swift (git pull), or run the example script to see if it runs properly?

**Comment by HongyongZZZ** [comment]
Due to the school campus network being unable to access GitHub these days, I may need a longer time to keep retrying.
I just created a new environment and installed Swift version 3.7.3 using pip, but the same issue occurs in this environment.

**Comment by Zzzlkh** [comment]
I'm facing the same problem. Have you managed to solve it?

**Comment by WeixuanXiong** [comment]
The same problem when using grpo training qwen3 series

**Comment by hjh0119** [comment]
please provide the repro script and environment

**Comment by WeixuanXiong** [comment]
export MASTER_ADDR=127.0.0.1
export MASTER_PORT=44150


CUDA_VISIBLE_DEVICES=0,1,2,3 \
NPROC_PER_NODE=4 \
swift rlhf \
    --rlhf_type grpo \
    --model_type qwen3_thinking \
    --model ./Qwen3-30B-A3B-Thinking-2507  \
    --external_plugins ./plugin.py \
    --reward_funcs len_orm timerange_orm miou_orm \
    --reward_weights 0.5 0.2 1 \
    --use_vllm true \
    --vllm_mode server \
    --vllm_server_host 127.0.0.1 \
    --vllm_server_port 8000 \
    --train_type full \
    --torch_dtype bfloat16 \
    --dataset ./grpo_data_clip_10742_20250904_promptv6.json \
    --max_length 16384 \
    --max_completion_length 2048 \
    --num_train_epochs 2 \
    --per_device_train_batch_size 8 \
    --per_device_eval_batch_size 8 \
    --learning_rate 5e-7 \
    --gradient_accumulation_steps 64 \
    --eval_steps 500 \
    --save_steps 30 \
    --save_total_limit 5 \
    --logging_steps 1 \
    --output_dir ./output \
    --warmup_ratio 0.01 \
    --dataloader_num_workers 4 \
    --num_generations 8 \
    --temperature 1.0 \
    --deepspeed zero3 \
    --log_completions true \
    --report_to wandb \
    --beta 0.001 \
    --num_iterations 1



Package                           Version        Editable project location
--------------------------------- -------------- -------------------------
absl-py                           2.3.1
accelerate                        1.7.0
addict                            2.4.0
aiofiles                          22.1.0
aiohappyeyeballs                  2.4.3
aiohttp                           3.11.7
aiosignal                         1.3.1
aiosqlite                         0.21.0
alabaster                         0.7.16
aliyun-python-sdk-core            2.16.0
aliyun-python-sdk-kms             2.16.5
annotated-types                   0.7.0
antlr4-python3-runtime            4.13.2
anyio                             4.6.2.post1
argon2-cffi                       25.1.0
argon2-cffi-bindings              25.1.0
arrow                             1.3.0
astor                             0.8.1
asttokens                         3.0.0
async-lru                         2.0.5
attrdict                          2.0.1
attrs                             24.2.0
audioread                         3.0.1
av                                13.1.0
babel                             2.16.0
beautifulsoup4                    4.13.5
bidict                            0.23.1
binpacking                        1.5.2
blake3                            1.0.5
bleach                            6.2.0
blinker                           1.5
Brotli                            1.1.0
cachetools                        6.2.0
cbor2                             5.7.0
certifi                           2024.8.30
cffi                              1.17.1
chardet                           5.1.0
charset-normalizer                3.4.0
click                             8.1.7
cloudpickle                       3.1.1
comm                              0.2.3
compressed-tensors                0.10.2
contourpy                         1.3.1
coverage                          7.10.6
cpm-kernels                       1.0.11
crcmod                            1.7
cryptography                      45.0.5
cupy-cuda12x                      13.6.0
cycler                            0.12.1
dacite                            1.9.2
datasets                          3.1.0
dbus-python                       1.3.2
debugpy                           1.8.16
decorator                         5.2.1
deepspeed                         0.16.8
defusedxml                        0.7.1
Deprecated                        1.2.18
depyf                             0.19.0
devscripts                        2.23.4+deb12u1
dill                              0.3.8
diskcache                         5.6.3
distlib                           0.4.0
distro                            1.8.0
distro-info                       1.5+deb12u1
dnspython                         2.7.0
docker-pycreds                    0.4.0
docstring_parser                  0.16
docutils                          0.19
e                                 1.4.5
einops                            0.8.0
email-validator                   2.3.0
entrypoints                       0.4
executing                         2.2.1
fastapi                           0.115.5
fastapi-cli                       0.0.10
fastapi-cloud-cli                 0.1.5
fastjsonschema                    2.21.2
fastrlock                         0.8.3
ffmpy                             0.4.0
filelock                          3.16.1
fire                              0.7.0
flash_attn                        2.8.2
fonttools                         4.55.0
fqdn                              1.5.1
frozenlist                        1.5.0
fsspec                            2024.9.0
future                            1.0.0
gevent                            24.11.1
gguf                              0.17.1
gitdb                             4.0.12
GitPython                         3.1.45
gpg                               1.18.0
gradio                            5.44.1
gradio_client                     1.12.1
greenlet                          3.2.4
groovy                            0.1.2
grpcio                            1.74.0
gunicorn                          20.1.0
h11                               0.16.0
hf_transfer                       0.1.9
hf-xet                            1.1.5
hjson                             3.1.0
httpcore                          1.0.9
httplib2                          0.20.4
httptools                         0.6.4
httpx                             0.28.1
huggingface-hub                   0.34.3
idna                              3.10
imagesize                         1.4.1
importlib_metadata                8.7.0
importlib_resources               6.4.5
iniconfig                         2.1.0
interegular                       0.3.3
iotop                             0.6
ipaddress                         1.0.23
ipykernel                         6.29.5
ipython                           9.5.0
ipython-genutils                  0.2.0
ipython_pygments_lexers           1.1.1
ipywidgets                        8.1.7
isoduration                       20.11.0
jedi                              0.19.2
jieba                             0.42.1
Jinja2                            3.1.4
jiter                             0.10.0
jmespath                          0.10.0
joblib                            1.4.2
json5                             0.12.1
jsonpatch                         1.33
jsonpointer                       3.0.0
jsonschema                        4.25.1
jsonschema-specifications         2025.4.1
jupyter                           1.1.1
jupyter-client                    7.0.0
jupyter-console                   6.6.3
jupyter_core                      5.8.1
jupyter-events                    0.12.0
jupyter-lsp                       2.3.0
jupyter_server                    2.17.0
jupyter_server_fileid             0.9.3
jupyter_server_terminals          0.5.3
jupyter_server_ydoc               0.8.0
jupyter-ydoc                      0.2.5
jupyterlab                        3.6.4
jupyterlab_pygments               0.3.0
jupyterlab_server                 2.27.3
jupyterlab_widgets                3.0.15
kiwisolver                        1.4.7
langchain                         0.3.27
langchain-core                    0.3.75
langchain-text-splitters          0.3.11
langsmith                         0.4.25
lark                              1.2.2
latex2sympy2_extended             1.0.6
lazr.restfulclient                0.14.5
lazr.uri                          1.0.6
lazy_loader                       0.4
librosa                           0.11.0
llamafactory                      0.9.4.dev0     /tmp/LLaMA-Factory
llguidance                        0.7.30
llvmlite                          0.44.0
lm-format-enforcer                0.10.12
Markdown                          3.9
markdown-it-py                    3.0.0
MarkupSafe                        2.1.5
math-verify                       0.5.2
matplotlib                        3.9.2
matplotlib-inline                 0.1.7
mdurl                             0.1.2
mistral_common                    1.8.4
mistune                           3.1.4
modelscope                        1.28.1
mpmath                            1.3.0
ms_swift                          3.8.0.dev0
msgpack                           1.0.8
msgspec                           0.19.0
multidict                         6.1.0
multiprocess                      0.70.16
nbclassic                         1.3.2
nbclient                          0.10.2
nbconvert                         7.16.6
nbformat                          5.10.4
nest-asyncio                      1.6.0
networkx                          3.4.2
ninja                             1.11.1.4
nltk                              3.9.1
none                              0.1.1
notebook                          6.5.7
notebook_shim                     0.2.4
numba                             0.61.2
numpy                             2.2.6
nvidia-cublas-cu12                12.6.4.1
nvidia-cuda-cupti-cu12            12.6.80
nvidia-cuda-nvrtc-cu12            12.6.77
nvidia-cuda-runtime-cu12          12.6.77
nvidia-cudnn-cu12                 9.5.1.17
nvidia-cufft-cu12                 11.3.0.4
nvidia-cufile-cu12                1.11.1.6
nvidia-curand-cu12                10.3.7.77
nvidia-cusolver-cu12              11.7.1.2
nvidia-cusparse-cu12              12.5.4.2
nvidia-cusparselt-cu12            0.6.3
nvidia-ml-py                      13.580.65
nvidia-nccl-cu12                  2.26.2
nvidia-nvjitlink-cu12             12.6.85
nvidia-nvtx-cu12                  12.6.77
nvitop                            1.5.3
oauthlib                          3.2.2
omegaconf                         2.3.0
openai                            1.106.1
openai-harmony                    0.0.4
opencv-python-headless            4.12.0.88
orjson                            3.10.12
oss2                              2.19.1
outlines_core                     0.2.10
overrides                         7.7.0
packaging                         24.2
pandas                            2.2.3
pandocfilters                     1.5.1
parso                             0.8.5
partial-json-parser               0.2.1.1.post6
pathlib2                          2.3.7.post1
pathtools                         0.1.2
peft                              0.15.2
pexpect                           4.8.0
pillow                            11.3.0
pip                               25.2
platformdirs                      4.3.8
pluggy                            1.6.0
ply                               3.11
pooch                             1.8.2
prometheus_client                 0.22.1
prometheus-fastapi-instrumentator 7.1.0
promise                           2.3
prompt_toolkit                    3.0.52
propcache                         0.2.0
protobuf                          3.20.3
psutil                            6.0.0
ptyprocess                        0.7.0
pure_eval                         0.2.3
py                                1.11.0
py-cpuinfo                        9.0.0
pyarrow                           18.0.0
pybase64                          1.4.2
pycountry                         24.6.1
pycparser                         2.22
pycryptodome                      3.23.0
pycryptodomex                     3.23.0
pydantic                          2.11.7
pydantic_core                     2.33.2
pydantic-extra-types              2.10.5
pydub                             0.25.1
Pygments                          2.18.0
PyGObject                         3.42.2
PyJWT                             2.6.0
pyOpenSSL                         25.1.0
pyparsing                         3.0.9
pytest                            6.2.5
pytest-cov                        6.2.1
pytest-cover                      3.0.0
pytest-coverage                   0.0
python-apt                        2.6.0
python-consul                     1.1.0
python-dateutil                   2.9.0.post0
python-debian                     0.1.49
python-dotenv                     1.1.1
python-engineio                   4.12.2
python-etcd                       0.4.5
python-json-logger                3.3.0
python-magic                      0.4.26
python-multipart                  0.0.20
python-socketio                   5.13.0
pytz                              2024.2
pyxdg                             0.28
PyYAML                            6.0.2
pyzmq                             27.0.2
ray                               2.49.1
referencing                       0.36.2
regex                             2024.11.6
requests                          2.32.3
requests-toolbelt                 1.0.0
rfc3339-validator                 0.1.4
rfc3986                           1.5.0
rfc3986-validator                 0.1.1
rfc3987-syntax                    1.1.0
rich                              13.9.4
rich-toolkit                      0.15.1
rignore                           0.6.4
rouge                             1.0.1
rouge-chinese                     1.0.3
rpds-py                           0.27.1
ruff                              0.12.12
safehttpx                         0.1.6
safetensors                       0.4.5
schedule                          1.2.2
scikit-learn                      1.7.1
scipy                             1.14.1
semantic-version                  2.10.0
Send2Trash                        1.8.3
sentencepiece                     0.2.0
sentry-sdk                        2.34.1
setproctitle                      1.3.6
setuptools                        80.9.0
shellingham                       1.5.4
shortuuid                         1.0.13
shtab                             1.7.1
simple-websocket                  1.1.0
simplejson                        3.20.1
six                               1.16.0
smmap                             5.0.2
sniffio                           1.3.1
snowballstemmer                   2.2.0
sortedcontainers                  2.4.0
soundfile                         0.13.1
soupsieve                         2.8
soxr                              0.5.0.post1
Sphinx                            5.3.0
sphinxcontrib-applehelp           2.0.0
sphinxcontrib-devhelp             2.0.0
sphinxcontrib-htmlhelp            2.1.0
sphinxcontrib-jsmath              1.0.1
sphinxcontrib-qthelp              2.0.0
sphinxcontrib-serializinghtml     2.0.0
sphinxcontrib-websupport          2.0.0
SQLAlchemy                        2.0.43
sse-starlette                     2.1.3
stack-data                        0.6.3
starlette                         0.41.3
sympy                             1.13.3
tenacity                          9.1.2
tensorboard                       2.20.0
tensorboard-data-server           0.7.2
termcolor                         2.5.0
terminado                         0.18.1
threadpoolctl                     3.6.0
thriftpy2                         0.5.3
tiktoken                          0.8.0
tinycss2                          1.4.0
tokenizers                        0.21.1
toml                              0.10.2
tomlkit                           0.12.0
torch                             2.7.1
torchaudio                        2.7.1
torchvision                       0.22.1
tornado                           6.5.2
tox                               3.28.0
tqdm                              4.67.1
traitlets                         5.14.3
transformers                      4.55.4
transformers-stream-generator     0.0.5
triton                            3.3.1
trl                               0.20.0
typer                             0.17.4
types-python-dateutil             2.9.0.20250822
typing_extensions                 4.12.2
typing-inspection                 0.4.1
tyro                              0.8.14
tzdata                            2024.2
unattended-upgrades               0.1
unidiff                           0.7.3
uri-template                      1.3.0
urllib3                           1.26.20
uvicorn                           0.32.1
uvloop                            0.21.0
virtualenv                        20.33.0
vllm                              0.10.1.1
wadllib                           1.3.6
watchdog                          6.0.0
watchfiles                        1.1.0
wcwidth                           0.2.13
webcolors                         24.11.1
webencodings                      0.5.1
websocket-client                  1.8.0
websockets                        12.0
Werkzeug                          3.1.3
wheel                             0.45.0
widgetsnbextension                4.0.14
wrapt                             1.17.2
wsproto                           1.2.0
xdg                               5
xformers                          0.0.31
xgrammar                          0.1.21
xxhash                            3.5.0
y-py                              0.6.2
yarl                              1.18.0
ypy-websocket                     0.8.4
zipp                              3.23.0
zope.event                        5.1.1
zope.interface                    7.2
zstandard                         0.24.0



will not stuck when training qwen3-4b-thinking

**Comment by hjh0119** [comment]
@WeixuanXiong 

It may be hanging due to slow weight synchronization in the MoE model; this issue will be optimized within one to two weeks

**Comment by liuxuexun** [comment]
数据路径有问题或者奖励函数有问题都会卡死，不会报相应的错误，优先检查一下这些
